src/api/syscall.c

```diff
@@ -58,14 +62,12 @@ exception_t handleInterruptEntry(void)
         handleSpuriousIRQ();
     }
 
-#ifdef CONFIG_KERNEL_MCS
+    // If the lock isn't held then schedule() and activateThread()
+    // aren't safe to call.
     if (SMP_TERNARY(clh_is_self_in_queue(), 1)) {
-#endif
         schedule();
         activateThread();
-#ifdef CONFIG_KERNEL_MCS
     }
-#endif
```

Comment:
We talked about this before and I think the conclusion was that this change should be done, but can't remember where. Anyway, I don't see why the behaviour should differ between MCS and non-MCS here.

IIRC, x86 virtualisation might depend on the schedule call here in case of cross-core signals, or something like that? Of course that would be broken for MCS now too, just something to keep in mind.